### DISCUSSION TITLE
help running on expo sdk 51

### DISCUSSION THREAD

**Discussion by VengasAlejandro** [rnmapbox/maps#3507] [Q&A]
sorry for bad English not first language 
I'm tiring to us this library on my expo project, but i get an error  while running on android about mapbox.setaccestoken not being a function and being undefined , but on my vs code it get recognized  and when i hover the function it says its a const witch is true.
 my question is, maybe my setup is not design for running this library?
 i have:
 the latest : https://expo.dev/changelog/2024/05-07-sdk-51
 project made with npx create-expo-app --template default@beta
 added typescript from the expo documentation instead of js and also eslint from the same changelog.
 also using expo router v3 and am not using expo go. 
 I'm building my own apk with npx expo run:android
 maybe this library is not updated to run with new react archuitecture and new way expo bundles their librarys? 
 ty for your time and effort on this amazing tool

**Comment by mfazekas** [comment]
The lib should work with expo 51. Please check our install guides - https://rnmapbox.github.io/docs/install

**Reply by bkamrani** [reply]
It works with sdk 51. I have explained the process I took to do to get it to work here. Hope that helps.  https://stackoverflow.com/questions/78600582/expo-eas-build-fails-due-to-mapbox-token-missing/78600583#78600583
